submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/api/FilesService.kt

```diff
@@ -1,8 +1,14 @@
 package ac.uk.ebi.biostd.persistence.filesystem.api
 
+import ebi.ac.uk.extended.model.ExtFile
 import ebi.ac.uk.extended.model.ExtSubmission
 
 internal interface FilesService {
-    fun persistSubmissionFiles(sub: ExtSubmission): ExtSubmission
+    fun preProcessSubmissionFiles(sub: ExtSubmission): FilePersistenceConfig
```

Comment:
I think siganutre of this methods should be

```
    fun preProcessSubmissionFiles(sub: ExtSubmission) {}
    fun persistSubmissionFile(sub: ExtSubmission, file: ExtFile): ExtFile
    fun postProcessSubmissionFiles(sub: ExtSubmission) {}
  ```
  
The problem that I see in current desing is that nothing is preventing for calling processing with the incorrect config type. With interfaces defenied as bellow we do not have that problem but on the other hand we need to recalculate on each file the target file. Personally I prefer safer code that "performance" code (calculating expecting folder is not heavy operation) but is open to discussion.

Reply:
yes exactly. I think in the pre process temp folder will be created